### ISSUE TITLE
TFT35 freezes before Purging

### ISSUE THREAD

**Issue by MrKuskov** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2895]
### Description
Biqu B1 Printer

### Steps to reproduce
I am sending a file for printing
[flowrate_0_PLA_7m36s.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/14345696/flowrate_0_PLA_7m36s.zip)
all G-code commands pass normally until the M117 Purging command line 291 in the file. Then the screen freezes. If you send it to print from Marlin mode, then the file is printed. But it is not possible to go back to "Touch mode". The mode does not switch when holding the knob. The same happens when printing any files.

**Expected behavior**

**Actual behavior**

### Hardware Variant
 GD_TFT35_V3.0_B1_V27.x
### TFT Firmware Version & Main Board Firmware details

The TFT firmware version from February 20th. From February 17, it worked fine. Marlin 2.1.2.2
### Additional Information
* Include a ZIP file containing your `Configuration.h` or use [Pastebin](https://pastebin.com/) and paste a link in this issue.
[Configuration.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/14345832/Configuration.zip)

* Provide pictures or links to videos that clearly demonstrate the issue.

**Comment by MrKuskov** [comment]
> If you are on the latest TFT firmware, do you have the new setting `command_checksum` enabled on TFT's `config.ini` file? eventually try to disable it (although it should not be related to the issue) and let me know the result. Thanks
Yes. The function is available, I tried to turn it on and off in the screen settings. my screen settings via the [link](https://github.com/MrKuskov/BIGTREETECH-TouchScreenFirmware/blob/master/Copy%20to%20SD%20Card%20root%20directory%20to%20update/config.ini)

> **EDIT:** I tested your gcode file on two printers without any issue. Before `M117 Purging` gcode the nozzle temp needs to reach its target. After it was reached, the print started and terminated without any issue

Yes, in Marlin mode, I also type without problems. Printing freezes after line 
`G1 X6.1 Y20 Z0.3 F5000.0; Move to start position`
 line 290 in the code. When the hotend starts moving to the specified line, the screen stops responding. The "M117 Purging" line is no longer being executed. The command is executed without problems via Pronterface.

**Comment by MrKuskov** [comment]
I found out that printing is interrupted in "Touch mode" only if the card is installed in the USB or SD of the screen. Printing is normal when installed in the SD board.

**Comment by digant73** [comment]
@MrKuskov try with another SD card, if not already done. I have no issue at all on my printers (5 prints without any problem)

**Comment by MrKuskov** [comment]
I flashed the version from January 23rd. https://github.com/MrKuskov/BIGTREETECH-TouchScreenFirmware/blob/2ae78153bf685e901e504a04a88d9b6021450057/Copy%20to%20SD%20Card%20root%20directory%20to%20update/BIGTREE_GD_TFT35_V3.0_B1.27.x.bin Everything works correctly in it.

**Comment by digant73** [comment]
@MrKuskov please attach here the `config.ini` file you are using

**Comment by MrKuskov** [comment]
No, it doesn't work. The screen froze when the hotend moved to the starting point, but the hotend executed three more commands: completed two cleaning lines and moved to the center of the table.

**Comment by digant73** [comment]
@ok thanks, I will continue to investigate. Could you please attach your `config.ini` file?  so I will try to use it

**Comment by digant73** [comment]
> I have made a fork from your repository and am making changes to it for both the marlin firmware and the screen firmware https://github.com/MrKuskov/BIGTREETECH-TouchScreenFirmware/blob/master/Copy%20to%20SD%20Card%20root%20directory%20to%20update/config.ini

ok. I will use your `config.ini` file to try to replicate the issue

**Comment by digant73** [comment]
@MrKuskov Even with your `config.ini` file I have no issue. the only thing I saw is that command M280 at line 282 was skipped by Marlin (I received the `Unknow Command` message) due it is not configured/supported on my printer.

**Comment by MrKuskov** [comment]
Sorry I made a mistake. The firmware version is from January 23rd. [This one](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/blob/2ae78153bf685e901e504a04a88d9b6021450057/Copy%20to%20SD%20Card%20root%20directory%20to%20update/BIGTREE_GD_TFT35_V3.0_B1.27.x.bin). I use the ru language pack. But I tried switching to the en version. The screen freezes the same way.

**Comment by MrKuskov** [comment]
@digant73 Yes!!! The `Event Led` function is to blame for the screen freezes. When it is turned off, everything works correctly. If the function is enabled during printing, the screen freezes immediately.

**Comment by digant73** [comment]
ok thanks. Do you have a LED strip configured in your printer or a simple LED on the hotend?

**Comment by MrKuskov** [comment]
Multi-color RGB lights from [Biqu B1](https://biqu.equipment/products/biqu-b1-3d-printer-skr-v1-4-32bit-motherboard-tft35-b1-v3-0-dual-operation-system-fdm-impressora-3d-printer-beginner?variant=31952915005538)

**Comment by digant73** [comment]
could you let me know which Marlin settings reported at the beginning of `config.ini` file (on section `Supported Marlin Firmware Versions`) are enabled and disabled on your Marlin?

**Comment by MrKuskov** [comment]
I have enabled all the settings specified at the beginning of the config.ini file in my Marlin
[Configuration.h](https://github.com/MrKuskov/Configurations/blob/release-2.1.2.2/config/examples/BIQU/B1%20-%20SKR%201.4/Configuration.h)
[Configuration_adv.h](https://github.com/MrKuskov/Configurations/blob/release-2.1.2.2/config/examples/BIQU/B1%20-%20SKR%201.4/Configuration_adv.h)
in my repositories

**Comment by MrKuskov** [comment]
> If the function `Event LED` is enabled during printing, the screen freezes immediately. 
But if I go to the LED menu, I can turn on any backlight colors and the screen works correctly.

**Comment by digant73** [comment]
Ok. I will check on `LED Event` code

**Comment by MrKuskov** [comment]
@digant73 everything is working correctly. very good. thanks

**Comment by digant73** [comment]
ok. but remember that I commented out just a piece of code to have a confirmation of the issue. I will provide a PR with the fix. Leave this bug report open, it will automatically closed once the PR is merged. Thanks for support

**Comment by MrKuskov** [comment]
> **EDIT:** Please if possible let me know if the attached fw is ok (that should fix the issue).

Everything works.

**Comment by digant73** [comment]
thanks. The fix was also merged in the official btt fw just today.
